### PR TITLE
Adds concealed weapon bay

### PR DESCRIPTION
**Pull request by swindly**
:cl: Swindly
add: Added concealed weapon bays. They allow a non-combat mecha to equip a mecha weapon. They can be purchased by traitor roboticists and research directors from the traitor uplink.
/:cl:

[why]: This provides roboticists with an item that broadens their play style by introducing a stealth element.

The concealed weapon bay is an item that attaches to Ripleys or Odysseuses and allows them to equip one mecha weapon. A mecha can only have one attached.  Examining the mecha will not reveal the attached weapon (although the raising message still shows). They cost 3 TC because they don't allow roboticists to make mechas much more dangerous than the ones they can already make and because roboticists will still need to have researched the weaponry to attach to the mechas.

I made the sprite all by myself.

### CODE REVIEW

**Review by nicbn — CHANGES_REQUESTED** [review]

**Comment by improvedname** [comment]
you can already emagg a ripley for the same effect

**Comment by swindly** [comment]
> you can already emagg a ripley for the same effect

No?

**Comment by GunHog** [comment]
Can mechs even BE emagged? I have never seen or heard of such a thing.

**Comment by swindly** [comment]
Mechas can not be emagged.

File: code/game/mecha/mecha_control_console.dm [hunk 1]
```diff
@@ -91,6 +91,12 @@
 			M.trackers -= src
 	return ..()
 
+/obj/item/mecha_parts/mecha_equipment/try_attach_part(mob/user, obj/mecha/M)
```
Comment (SpaceManiac) [inline code/game/mecha/mecha_control_console.dm]: Should this be `mecha_tracking` rather than `mecha_equipment`?

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/game/mecha/mecha_control_console.dm (hunk 1 above)
Comment (swindly) [reply code/game/mecha/mecha_control_console.dm]: yes

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 2]
```diff
@@ -10,11 +10,14 @@
...
 	var/kickback = TRUE //Will using this weapon in no grav push mecha back.
 
-/obj/item/mecha_parts/mecha_equipment/weapon/can_attach(obj/mecha/combat/M)
-	if(..())
-		if(istype(M))
-			return 1
-	return 0
+/obj/item/mecha_parts/mecha_equipment/weapon/can_attach(obj/mecha/M)
+	if(!..())
+		return
```
Comment (nicbn) [inline code/game/mecha/equipment/weapons/weapons.dm]: I think that should be `return FALSE`?

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**